tests/target/issue-1824.rs

```diff
@@ -0,0 +1,181 @@
+use std::collections::{HashMap, HashSet};
```

Comment:
It would be better I think to have a minimal test case for this issue - can you cut this down to a few lines?